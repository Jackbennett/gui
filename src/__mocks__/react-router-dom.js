import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
