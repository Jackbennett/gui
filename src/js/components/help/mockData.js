export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
