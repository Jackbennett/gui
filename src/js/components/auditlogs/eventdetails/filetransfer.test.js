import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { defaultState, undefineds } from '../../../../../tests/mockData';
import FileTransfer from './filetransfer';

const mockStore = configureStore([thunk]);

describe('FileTransfer Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Provider store={store}>
          <FileTransfer item={{ ...defaultState.organization.events[2], meta: { path: ['/dev/null'] } }} />
        </Provider>
      </MemoryRouter>
    );

    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
