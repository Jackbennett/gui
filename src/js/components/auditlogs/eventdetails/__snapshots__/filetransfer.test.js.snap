// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileTransfer Component renders correctly 1`] = `
<div
  class="flexbox column"
  style="margin: 24px; min-width: min-content;"
>
  <div
    class="flexbox column margin-top-small"
  >
    <b
      class="margin-bottom-small capitalized-start"
    >
      device
       details
    </b>
    <div
      class="text-muted two-columns"
      style="grid-template-columns: minmax(max-content, 150px) max-content; row-gap: 20px;"
    >
      <div
        class="align-right"
      >
        <b>
          Device ID
        </b>
      </div>
      <div>
        <a
          class="flexbox"
          href="/devices?id=a1"
          style="align-items: center; color: rgba(0, 0, 0, 0.54); font-weight: initial;"
        >
          <span>
            a1
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root margin-left-small link-color MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </a>
      </div>
      <div
        class="align-right"
      >
        <b>
          Device type
        </b>
      </div>
      <div>
        raspberrypi4
      </div>
      <div
        class="align-right"
      >
        <b>
          System software version
        </b>
      </div>
      <div>
        -
      </div>
      <div
        class="align-right"
      >
        <b>
           
        </b>
      </div>
      <div>
        <a
          href="/auditlog?object_type=device&object_id=a1&start_date=2020-01-01T00:00:00.000Z"
        >
          List all log entries for this device
        </a>
      </div>
    </div>
  </div>
  <div
    class="flexbox column margin-top-small"
  >
    <b
      class="margin-bottom-small capitalized-start"
    >
      file transfer
       details
    </b>
    <div
      class="text-muted two-columns"
      style="grid-template-columns: minmax(max-content, 150px) max-content; row-gap: 20px;"
    >
      <div
        class="align-right"
      >
        <b>
          Path
        </b>
      </div>
      <div>
        /dev/null
      </div>
      <div
        class="align-right"
      >
        <b>
          User
        </b>
      </div>
      <div>
        string@example.com
      </div>
    </div>
  </div>
</div>
`;
